vspace/src/main/java/edu/asu/diging/vspace/web/staff/SpaceOverviewController.java

```diff
@@ -0,0 +1,172 @@
+            } else {
+                List<ISpaceLinkDisplay> tempSpaceToSpacelinkList = AllSpaceToSpacelinksListMap
+                        .get(link.getLink().getSourceSpace().getId());
+                tempSpaceToSpacelinkList.add(link);
+                AllSpaceToSpacelinksListMap.put(link.getLink().getSourceSpace().getId(), tempSpaceToSpacelinkList);
+            }
+        });
+
+        List<SpaceLinkDisplay> AllSpaceLinkForGivenOrNullSpaceStatusList = spaceLinkDisplayRepository
+                .findAllSpaceLinksForGivenOrNullSpaceStatus(SpaceStatus.PUBLISHED);
+        Map<String, List<ISpaceLinkDisplay>> AllSpaceLinkForGivenOrNullSpaceStatustMap = new HashMap<>();
+
+        AllSpaceLinkForGivenOrNullSpaceStatusList.forEach(link -> {
+            if (!AllSpaceLinkForGivenOrNullSpaceStatustMap.containsKey(link.getLink().getSourceSpace().getId())) {
+                List<ISpaceLinkDisplay> tempSpaceToSpacelinkList = new ArrayList<>();
+                tempSpaceToSpacelinkList.add(link);
+                AllSpaceLinkForGivenOrNullSpaceStatustMap.put(link.getLink().getSourceSpace().getId(),
+                        tempSpaceToSpacelinkList);
+            } else {
+                List<ISpaceLinkDisplay> tempSpaceToSpacelinkList = AllSpaceLinkForGivenOrNullSpaceStatustMap
+                        .get(link.getLink().getSourceSpace().getId());
+                tempSpaceToSpacelinkList.add(link);
+                AllSpaceLinkForGivenOrNullSpaceStatustMap.put(link.getLink().getSourceSpace().getId(),
+                        tempSpaceToSpacelinkList);
+            }
+        });
+        if (spaceList != null && !spaceList.isEmpty()) {
+            spaceList.forEach(s -> {
+                ObjectNode node = mapper.createObjectNode();
+                node.put("name", s.getName());
+                node.put("link", request.getContextPath() + "/staff/space/" + s.getId());
+                node.put("img", request.getContextPath() + "/api/image/" + s.getImage().getId());
+                node.put("isModule", false);
+                List<String> spaceAllLink = new ArrayList<>();
+                List<IModuleLinkDisplay> spaceToModulelinksList = AllSpaceToModulelinksListMap.get(s.getId());
+                List<ISpaceLinkDisplay> spaceLinks = null;
+                if (s.isShowUnpublishedLinks()) {
+                    spaceLinks = AllSpaceToSpacelinksListMap.get(s.getId());
+                } else {
+                    spaceLinks = AllSpaceLinkForGivenOrNullSpaceStatustMap.get(s.getId());
+                }
+                List<ISpaceLinkDisplay> spaceToSpaceLinksList = null;
+                if (spaceLinks != null) {
+                    spaceToSpaceLinksList = spaceLinks.stream().filter(spaceLinkDisplayObj -> !spaceLinkDisplayObj
+                            .getLink().getTargetSpace().isHideIncomingLinks()).collect(Collectors.toList());
```

Comment:
same here, the graph should show how all spaces and modules are connected and have icons to indicate if a space hides incoming links.